Pricing/ItemPriceCalculator.php

```diff
@@ -50,23 +50,50 @@ public function calculate($item, $currency = null, $useProductsPrice = true)
         // get bulk price
         if ($useProductsPrice) {
             $product = $item->getCalcProduct();
-            $price = $this->priceManager->getBulkPriceForCurrency($product, $item->getCalcQuantity(), $currency);
+
+            //get special price
+            $specialPrice = $this->priceManager->getSpecialPriceForCurrency($product, $currency);
+            if (is_object($specialPrice)) {
```

Comment:
i  guess checking for an object is not really necessary
and try avoiding the else.
````
$specialPriceValue = null;
if ($specialPrice) {
    $specialPriceValue = $specialPrice->getPrice();
}
````